src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -738,75 +920,81 @@
 
 (defn create-extract-node
   "Returns a `query-rec` that has the correct projection for the given
-  `column-list`. Updating :project causes the select in the SQL query
-  to be modified. Setting :late-project does not affect the SQL, but
+  `column-list`. Updating :projected-fields causes the select in the SQL query
+  to be modified. Setting :late-projected-fields does not affect the SQL, but
   includes in the information for later removing the columns."
   [query-rec column-list expr]
-  (let [project-map (zipmap column-list (repeat (count column-list) nil))]
-    (if (or (nil? expr)
-            (not (subquery-expression? expr)))
-      (let [qr (assoc query-rec :where (user-node->plan-node query-rec expr))]
-        (if (:supports-extract? query-rec)
-          (assoc qr :project project-map)
-          (assoc qr :late-project project-map)))
-      (let [[subquery-name & subquery-expression] expr]
-        (assoc (user-query->logical-obj subquery-name)
-          :project project-map
-          :where (when (seq subquery-expression)
-                   (user-node->plan-node (user-query->logical-obj subquery-name)
-                                         (first subquery-expression))))))))
+  (if (or (nil? expr)
+          (not (subquery-expression? expr)))
```

Comment:
So this was a problem before as well but when I changed everything to use `late-project?` I didn't add `late-project? false` to the appropriate query-recs because `late-project? nil` would also be fine for the if statement below. The problem is that now those query-rec's aren't instances of the Query protocol because they don't have a `late-project? false` field. That means that the queries which actually do support projections to the sql won't. I think we just want to add `late-project? true` to all the query-recs.